drivers/pwm/pwm_realtek_rts5912.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2025 Realtek, SIBG-SD7
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_pwm
+
+#include <soc.h>
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_rts5912.h>
+
+#include "reg/reg_pwm.h"
+
+LOG_MODULE_REGISTER(pwm, CONFIG_PWM_LOG_LEVEL);
+
+#define PWM_CYCLE_PER_SEC MHZ(50)
+
+struct pwm_rts5912_config {
+	uint32_t pwm_base;
```

Comment:
Use the `struct pwm_regs` type directly to avoid casting every time you use this.  This also needs to be declared volatile to prevent compiler reorderng.

```suggestion
	volatile struct pwm_regs *pwm_regs;
```

Reply:
done